clippy_lints/src/transmuting_null.rs

```diff
@@ -0,0 +1,101 @@
+use crate::utils::{match_qpath, paths, span_lint};
+use crate::consts::{constant_context, Constant};
+use if_chain::if_chain;
+use rustc::lint::{LintArray, LintPass, LateLintPass, LateContext};
+use rustc::{declare_tool_lint, lint_array};
+use rustc::hir::{Expr, ExprKind};
+use syntax::ast::{LitKind};
+
+declare_clippy_lint! {
+    pub TRANSMUTING_NULL,
+    correctness,
+    "transmutes from a null pointer to a reference, which is Undefined Behavior"
+}
+
+#[derive(Copy, Clone)]
+pub struct TransmutingNullPass;
+
+impl LintPass for TransmutingNullPass {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(
+            TRANSMUTING_NULL,
+        )
+    }
+
+    fn name(&self) -> &'static str {
+        "TransmutingNull"
+    }
+}
+
+const LINT_MSG : &str = "transmuting a known null pointer into a reference.";
+
+impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TransmutingNullPass {
+        
+    fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr) {
+
+        if_chain! {
+            if let ExprKind::Call(ref func, ref args) = expr.node;
+            if let ExprKind::Path(ref path) = func.node;
+            if match_qpath(path, &["std", "mem", "transmute"]);
+            if args.len() == 1;
+            
+            then {
+
```

Comment:
I think you also have to run `rustfmt`.

Reply:
I think my `rustfmt` setup broke after I started using the `master` toolchain. Trying to run it from the console, that seems to be the case. I get these errors:

```console
$ cargo fmt
error: 'cargo-fmt' is not installed for the toolchain 'master'
To install, run `rustup component add rustfmt`
$ rustup component add rustfmt
error: toolchain 'master' does not support components
info: caused by: invalid toolchain name: 'master'
```

Do you know how to use `rustfmt` with the `master` toolchain?

Reply:
You'll have to use rustfmt from nighty as tools are not available on the master toolchain

Reply:
FYI: There is also 
```
-c, --component <components>...      additional components to install, besides rustc and rust-std
```
on RTIM, but a few months ago I didn't get this to work. Maybe now it works. But the CI runs nightly rustfmt anyway.